server/src/main/scala/actors/minutes/MinutesActorLike.scala

```diff
@@ -43,58 +50,85 @@ abstract class MinutesActorLike[A, B <: WithTimeAccessor](now: () => SDateLike,
 
     case StreamFailure(t) => log.error(s"Stream failed", t)
 
-    case GetPortState(startMillis, endMillis) =>
-      val replyTo = sender()
-      val eventualMinutesForAllTerminals = terminals.map { terminal =>
-        handleLookups(terminal, SDate(startMillis), SDate(endMillis))
-      }
-      combineEventualMinutesContainers(eventualMinutesForAllTerminals).pipeTo(replyTo)
+    case PointInTimeQuery(pit, GetStateForDateRange(startMillis, endMillis)) =>
```

Comment:
we could do this using composition rather than inheritance by putting these partial functions into a shared location.

Reply:
noted